offchain-modules/packages/x/src/db/kv.ts

```diff
@@ -0,0 +1,29 @@
+import { Connection, Repository } from 'typeorm';
+import { KV } from './entity/kv';
+
+export class KVDb {
+  private kvDb: Repository<KV>;
+  constructor(private connection: Connection) {
+    this.kvDb = connection.getRepository(KV);
+  }
+  async get(key: string): Promise<KV[]> {
+    return this.kvDb.find({
+      where: {
+        key,
+      },
+      take: 1,
+    });
+  }
+
+  async set(key: string, value: string) {
+    const updateRes = await this.kvDb
+      .createQueryBuilder()
+      .update()
+      .set({ value: value })
+      .where('key = :key', { key: key })
+      .execute();
+    if (updateRes.affected === 0) {
+      await this.kvDb.save({ key: key, value: value });
```

Comment:
Split the 2 actions may have concurrent issue.
Check method in https://github.com/typeorm/typeorm/issues/1090#issuecomment-634391487.

Reply:
fixed